### PR TITLE
Add pretty focus states!

### PR DESCRIPTION
**Pull request by notwaldorf**
This PR is larger than it should be, because it turns out I didn't commit the `demo` and `test` folders with the last PR. Mistakes were made, here we are.

Also adds an inky focus state!

### CODE REVIEW

**Comment by notwaldorf** [comment]
@cdata @morethanreal  PTAL

File: demo/index.html [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  <head>
+    <title>paper-toggle-button demo</title>
+
+    <meta name="viewport" content="width=device-width, minimum-scale=1.0, initial-scale=1, user-scalable=yes">
+    <meta name="mobile-web-app-capable" content="yes">
+    <meta name="apple-mobile-web-app-capable" content="yes">
+
+    <script src="../../webcomponentsjs/webcomponents-lite.js"></script>
+    <link href="../../paper-styles/layout.html" rel="import">
+    <link href="../paper-toggle-button.html" rel="import">
```
Comment (cdata) [inline demo/index.html]: What is this crazy mirror world where `rel` attribute goes after `href` attribute? I will not stand for it :see_no_evil:

File: demo/index.html [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        --paper-toggle-button-checked-button-color: #ff4081;
+        --paper-toggle-button-checked-ink-color: #ff4081;
+      }
+    </style>
+  </head>
+  <body>
+    <section>
+      <div class="center horizontal layout">
+        <div class="flex">Wi-Fi</div>
+        <paper-toggle-button checked role="checkbox" something></paper-toggle-button>
```
Comment (cdata) [inline demo/index.html]: `something` :game_die:

**Comment by cdata** [comment]
WFM :+1:

**Comment by notwaldorf** [comment]
I'm going to hold off merging this until https://github.com/PolymerElements/iron-behaviors/pull/4 is merged, since it depends on it.

**Comment by morethanreal** [comment]
Is the focus state implemented in `Polymer.PaperRadioButtonBehavior`? And does this need tests for the focus states?

**Comment by notwaldorf** [comment]
@morethanreal Yes, it is. I was thinking about renaming it to `PaperInkyFocusButtonBehavior`, if that would make it more obvious. The `iron` behaviour has a test that focus works, but none of the elements do. Would they test that the ink is being displayed? Should that maybe be a test in the behaviour itself?

**Comment by morethanreal** [comment]
Yes, I was thinking about some kind of test that ensures the focus state is displayed and has the "right" appearance. If they can be generalized in the behavior that would be the right place for it, otherwise there should be tests in the elements.

**Comment by notwaldorf** [comment]
@morethanreal filed https://github.com/PolymerElements/paper-behaviors/issues/4. On it! :runner:

**Pull request merged**